data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v2/beacon/GetAttestationsV2Test.java

```diff
@@ -0,0 +1,120 @@
+import tech.pegasys.teku.spec.TestSpecInvocationContextProvider;
+import tech.pegasys.teku.spec.datastructures.metadata.ObjectAndMetaData;
+import tech.pegasys.teku.spec.datastructures.operations.Attestation;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitionCache;
+import tech.pegasys.teku.spec.util.DataStructureUtil;
+
+@TestSpecContext(allMilestones = true)
+public class GetAttestationsV2Test extends AbstractMigratedBeaconHandlerTest {
+
+  private SpecMilestone specMilestone;
+
+  @BeforeEach
+  void setup(final TestSpecInvocationContextProvider.SpecContext specContext) {
+    spec = specContext.getSpec();
+    dataStructureUtil = new DataStructureUtil(spec);
+    specMilestone = specContext.getSpecMilestone();
+    setHandler(new GetAttestationsV2(nodeDataProvider, new SchemaDefinitionCache(spec)));
+  }
+
+  @TestTemplate
+  public void shouldReturnAttestationsWhenFound() throws JsonProcessingException {
+    final List<Attestation> attestations =
+        List.of(dataStructureUtil.randomAttestation(), dataStructureUtil.randomAttestation());
+    final ObjectAndMetaData<List<Attestation>> attestationsAndMetaData =
+        new ObjectAndMetaData<>(
+            attestations, spec.getGenesisSpec().getMilestone(), false, false, false);
+    when(nodeDataProvider.getAttestationsAndMetaData(any(), any()))
+        .thenReturn(attestationsAndMetaData);
+
+    handler.handleRequest(request);
+    assertThat(request.getResponseCode()).isEqualTo(SC_OK);
+    assertThat(request.getResponseBody()).isEqualTo(attestationsAndMetaData);
+  }
+
+  @TestTemplate
+  public void shouldReturnEmptyListWhenNoAttestations() throws JsonProcessingException {
+    final ObjectAndMetaData<List<Attestation>> attestationsAndMetaData =
+        new ObjectAndMetaData<>(
+            Collections.emptyList(), spec.getGenesisSpec().getMilestone(), false, false, false);
+    when(nodeDataProvider.getAttestationsAndMetaData(any(), any()))
+        .thenReturn(attestationsAndMetaData);
+
+    handler.handleRequest(request);
+    assertThat(request.getResponseCode()).isEqualTo(SC_OK);
+    assertThat(request.getResponseBody()).isEqualTo(attestationsAndMetaData);
+  }
+
+  @TestTemplate
+  void metadata_shouldHandle400() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_BAD_REQUEST);
+  }
+
+  @TestTemplate
+  void metadata_shouldHandle500() throws JsonProcessingException {
+    verifyMetadataErrorResponse(handler, SC_INTERNAL_SERVER_ERROR);
+  }
+
+  @TestTemplate
+  void metadata_shouldHandle200() throws Exception {
+    List<Attestation> attestations;
+    if (specMilestone.isGreaterThanOrEqualTo(ELECTRA)) {
+      attestations =
+          List.of(
+              dataStructureUtil.randomElectraAttestation(),
+              dataStructureUtil.randomElectraAttestation());
+    } else {
+      attestations =
+          List.of(dataStructureUtil.randomAttestation(), dataStructureUtil.randomAttestation());
+    }
```

Comment:
wait, so we cant use `dataStrucutreUtil.randomAttestation` after electra? thats weird....

Reply:
I understand that you could if you were initializing datastructureUtil until with an Electra spec, because `randomAttestation()` uses `spec.getGenesisSchemaDefinitions()`.

But in the case you have datastrauctureUtil using a pre-Electra spec, you might need a way to say that you want Phase0 or Electra attestations.

Reply:
create dataStructureUtil with electra spec and use that, but we can't follow this model and have our templating continue to work happily, please don't introduce this, it wont work.

Reply:
Right, the DataStructureUtil uses the spec it's been initialised with and this case it will use the correct one. So no need for this distinction in the end. Removed it